backend/capellacollab/settings/modelsources/git/routes.py

```diff
@@ -1,62 +1,87 @@
 # SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
 # SPDX-License-Identifier: Apache-2.0
 
+import typing as t
 
 from fastapi import APIRouter, Depends, HTTPException
 from sqlalchemy.orm import Session
 
-from capellacollab.core.authentication.database import verify_admin
+from capellacollab.core.authentication.database import RoleVerification
 from capellacollab.core.authentication.jwt_bearer import JWTBearer
 from capellacollab.core.database import get_db
 from capellacollab.settings.modelsources.git import crud
 from capellacollab.settings.modelsources.git.core import get_remote_refs
+from capellacollab.settings.modelsources.git.injectables import (
+    get_existng_git_setting,
+)
 from capellacollab.settings.modelsources.git.models import (
+    DB_GitSettings,
     GetRevisionModel,
     GetRevisionsResponseModel,
     GitSettings,
+    GitSettingsGitGetResponse,
 )
+from capellacollab.users.models import Role
 
 router = APIRouter()
 
 
-def check_git_settings_instance_exists(
-    db: Session,
-    id: int,
-):
-    instance = crud.get_git_settings(db, id)
-    if not instance:
-        raise HTTPException(
-            status_code=404,
-            detail={
-                "reason": f"The git instance with id {id} does not exist."
-            },
-        )
-
-
-@router.get("/")
+@router.get(
+    "/",
+    response_model=t.List[GitSettingsGitGetResponse],
```

Comment:
Rather use `list[GitSettingsGitGetResponse]` with `from __future__ import annotations` at the beginning of the file than `t.List[GitSettingsGitGetResponse]`.

Reply:
I changed it to `list` here and in another route.